src/status_im2/subs/contact.cljs

```diff
@@ -251,4 +251,32 @@
                (assoc acc address contact)
                acc))
            {}
-           contacts)))
\ No newline at end of file
+           contacts)))
+
+(re-frame/reg-sub
+ :contacts/filtered-active-sections
+ :<- [:contacts/active]
+ :<- [:contacts/search-query]
+ (fn [[contacts query]]
+   (let [data (atom {})]
+     (doseq [i (range (count contacts))]
+       (let [first-char (get (:alias (nth contacts i)) 0)]
+         (when (or (empty? query) (str/includes? (str/lower-case (:alias (nth contacts i))) (str/lower-case query)))
+           (if-not (contains? @data first-char)
+             (swap! data #(assoc % first-char {:title first-char :data [(nth contacts i)]}))
+             (swap! data #(assoc-in % [first-char :data] (conj (:data (get @data first-char)) (nth contacts i))))))))
+     (swap! data #(sort @data))
+     (vals @data))))
+
+(re-frame/reg-sub
+ :contacts/group-members-sections
+ :<- [:contacts/current-chat-contacts]
+ (fn [members]
+   (let [admins  (filter :admin? members)
+         online  (filter #(and (not (:admin? %)) (:online? %)) members)
+         offline (filter #(and (not (:admin? %)) (not (:online? %))) members)]
+     (vals (cond-> {}
```

Comment:
you could do it in one go in reduce. now you have 3 filters for the same list and also 3 conditions